### DISCUSSION TITLE
Grooving operation

### DISCUSSION THREAD

**Discussion by Festivejelly** [kachurovskiy/nanoels#158] [Ideas]
it would be nice to have an operation for grooving.

Right now im just using the face operation to do it. but for internal grooves its a bit faff to set up as again it uses the soft endstops as the location to cut from and to.

Ideally we would set the groove depth and length and it would go from the software end stops and return to the endstops when done. Rather than using the end stops as the cutting params.

**Comment by kachurovskiy** [comment]
Nicholas, I guess you could also use the internal threading mode with a small pitch to make grooves.

End stops is how we specify limits for all modes. Could be misleading for the user to suggest them to specify the length instead since we don't really know the width of the tool hence can't promise the length of the cut?
